theseus/embodied/robot/link.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Optional, Any
+import torch
+
+
+class Link(abc.ABC):
+    def __init__(
+        self,
+        name: str,
+        id: int = -1,
+        parent: Any = None,
+        children: Optional[List[Any]] = None,
```

Comment:
Each link is connected to exactly one joint that connects to one parent link (except the root), and 0~N joints that connect to children links.
I find the Pybullet convention to be rather clean: Associate each link with the joint that connects to its parent.
So I suggest that we can have `Link._parent` and `Link._children` contain adjacent Links, which makes traversal for kinematics/dynamics easier instead of having to go through link-joint-link every time. If we want to access the joint that connects to some link, it can be done by something like `Link._joint` or `Link._parent_joint`.

Reply:
That's a good point. Thanks!